esp/esp/program/modules/handlers/resourcemodule.py

```diff
@@ -357,28 +357,29 @@ def resources_classroom_import(self, request, tl, one, two, module, extra, prog)
                 for i in range(min(len(ts_old), len(ts_new))):
                     ts_map[ts_old[i].id] = ts_new[i]
 
-                #   Iterate over the resources in the previous program
-                for res in past_program.getClassrooms():
-                    furnishing_dict[res.name] = set()
-                    #   If we know what timeslot to put it in, make a copy
-                    if res.event.id in ts_map:
-                        new_res = Resource(
-                            name = res.name,
-                            res_type = res.res_type,
-                            num_students = res.num_students,
-                            is_unique = res.is_unique,
-                            user = res.user,
-                            event = ts_map[res.event.id]
-                        )
-                        #   Check to avoid duplicating rooms (so the process is idempotent)
-                        if import_mode == 'save' and not Resource.objects.filter(name=new_res.name, event=new_res.event).exists() and str(res.id) in to_import:
-                            new_res.save()
-                        else:
-                            new_res.old_id = res.id
-                        if import_furnishings:
-                            new_furns = self.furnishings_import(res, new_res, self.program, import_mode, to_import)
-                            furnishing_dict[res.name].update(new_furn.res_type.name + (" (Hidden)" if new_furn.res_type.hidden else "") + ((": " + new_furn.attribute_value) if new_furn.attribute_value else "") for new_furn in new_furns)
-                        resource_list.append(new_res)
+                #   Iterate over the classrooms in the previous program
+                for resource in past_program.groupedClassrooms():
+                    for event in resource.timegroup:
+                        furnishing_dict[resource.name] = set()
```

Comment:
Is it right for this to be inside the inner ("event") loop?  It seem likes this will reset the furnishing_dict for this resource.name for every item in the inner loop.  And the furnishing_dict doesn't get used inside the loop anywhere (only updated on line 381), so something seems off.  In the "complete availability" section, this line is inside the outer loop.

Reply:
`furnishing_dict` is only used to show the furnishings in the import preview, and it's at the classroom level, so you're probably right here. And since it's a set, we don't need to worry about duplicates from line 381.